File: assignment_2/ADTs/IList.java
package ADTs;

import Exceptions.EmptyContainerException;
import Model.ProgramState;

public interface IList<T> {
    void addFirst(T elem);
    boolean isEmpty();
    T remove() throws EmptyContainerException;
    String toString();
    T get(int index);
    T getFirst();
}

File: Toy_Language_Interpreter_C/ADTs/IList.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace a7
{
    interface IList<T>
    {
        void AddFirst(T elem);
        bool IsEmpty();
        T Remove();
        String ToString();
    }
}
